.script/tests/KqlvalidationsTests/JsonFilesTestData/DataConnectorFilesLoader.cs

```diff
@@ -25,53 +24,84 @@ protected override List<string> GetDirectoryPaths()
+                            JSchema dataConnectorJsonSchema = JSchema.Parse(File.ReadAllText("DataConnectorSchema.json"));
 
-                                var jsonString = File.ReadAllText(filePath);
-                                JObject dataConnectorJsonObject = JObject.Parse(jsonString);
+                            var jsonString = File.ReadAllText(filePath);
+                            JObject dataConnectorJsonObject = JObject.Parse(jsonString);
 
-                                if (dataConnectorJsonObject.IsValid(dataConnectorJsonSchema))
-                                {
-                                    validFiles.Add(filePath);
-                                }
-                                else
-                                {
-                                    throw new Exception("Invalid JSON schema for file: " + filePath);
-                                }
-                            }
-                            catch (JsonReaderException ex)
+                            if (dataConnectorJsonObject.IsValid(dataConnectorJsonSchema))
                             {
-                                Console.WriteLine("Invalid JSON file: " + filePath);
-                                Console.WriteLine("Error message: " + ex.Message);
+                                validFiles.Add(filePath);
                             }
-                            catch (Exception ex)
+                            else
                             {
-                                Console.WriteLine("An error occurred while processing file: " + filePath);
-                                Console.WriteLine("Error message: " + ex.Message);
+                                throw new Exception("Invalid JSON schema for file: " + filePath);
                             }
-                        });
-                    }
-                    else
-                    {
-                        Console.WriteLine("No JSON files found in directory: " + directoryPath);
+                        }
+                        catch (JsonReaderException ex)
+                        {
+                            Console.WriteLine("Invalid JSON file: " + filePath);
+                            Console.WriteLine("Error message: " + ex.Message);
+                        }
+                        catch (Exception ex)
+                        {
+                            Console.WriteLine("An error occurred while processing file: " + filePath);
+                            Console.WriteLine("Error message: " + ex.Message);
+                        }
                     }
-
-                    return accumulator.Concat(validFiles).ToList();
                 });
+
+                if (validFiles.Count == 0)
+                {
+                    validFiles.Add("NoFile.json");
+                }
+
+                return validFiles;
             }
             catch (Exception ex)
```

Comment:
We should remove nested try catch. Instead, we can have in each part of the code.

Reply:
I have added nested try catch just to make sure that if any single file having issue, then we can log the exception and continue the flow for other flows.